pgrx-tests/src/framework.rs

```diff
@@ -420,26 +488,42 @@ fn initdb(postgresql_conf: Vec<&'static str>) -> eyre::Result<()> {
 }
 
 fn modify_postgresql_conf(pgdata: PathBuf, postgresql_conf: Vec<&'static str>) -> eyre::Result<()> {
-    let mut postgresql_conf_file = std::fs::OpenOptions::new()
-        .write(true)
-        .truncate(true)
-        .open(format!("{}/postgresql.auto.conf", pgdata.display()))
-        .wrap_err("couldn't open postgresql.auto.conf")?;
-    postgresql_conf_file
-        .write_all("log_line_prefix='[%m] [%p] [%c]: '\n".as_bytes())
-        .wrap_err("couldn't append log_line_prefix")?;
+    let mut contents = String::new();
 
+    contents.push_str("log_line_prefix='[%m] [%p] [%c]: '\n");
+    contents
+        .push_str(&format!("unix_socket_directories = '{}'\n", pgdata.parent().unwrap().display()));
     for setting in postgresql_conf {
-        postgresql_conf_file
-            .write_all(format!("{setting}\n").as_bytes())
-            .wrap_err("couldn't append custom setting to postgresql.conf")?;
+        contents.push_str(&format!("{setting}\n"));
+    }
+
+    let postgresql_auto_conf = pgdata.join("postgresql.auto.conf");
+
+    if let Some(runas) = get_runas() {
+        let mut sudo_command = Command::new("sudo")
+            .arg("-u")
+            .arg(&runas)
+            .arg("tee")
```

Comment:
You mentioned making a `SudoCommand` type and I think what I'd like to *actually* see is these ~4 lines made into a function/whatever that returns `Command`.